scipy/stats/stats.py

```diff
@@ -3692,6 +3695,136 @@ def f_oneway(*args, axis=0):
+           of Educational Statistics, vol. 19, no. 2, 1994, pp. 91-101.
+           JSTOR, www.jstor.org/stable/1165140. Accessed 12 Sept. 2020.
+
+    Examples
+    --------
+        >>> from scipy.stats import alexandergovern
+
+        Here are some data on annual percentage rate of interest charged on
+        new car loans at nine of the largest banks in 4 American cities.
+        https://www.itl.nist.gov/div898/education/datasets.htm#anova
+
+        >>> atlanta = [13.75, 13.75, 13.5, 13.5, 13.0, 13.0, 13.0, 12.75, 12.5]
+        >>> chicago = [14.25, 13.0, 12.75, 12.5, 12.5, 12.4, 12.3, 11.9, 11.9]
+        >>> houston = [14.0, 14.0, 13.51, 13.5, 13.5, 13.25, 13.0, 12.5, 12.5]
+        >>> memphis = [15.0, 14.0, 13.75, 13.59, 13.25, 12.97, 12.5, 12.25,
+        ...           11.89]
+        >>> alexandergovern(atlanta, chicago, houston, memphis)
+        AlexanderGovernResult(statistic=4.65087071883494,
+                              pvalue=0.19922132490385214)
+    """
+
+    args = _alexandergovern_input_validation(args)
+
+    # The following formula numbers reference the equation described on
+    # page 92 by Alexander, Govern. Formulas 5, 6, and 7 describe other
+    # tests that serve as the basis for equation (8) but are not needed
+    # to perform the test.
+
+    # precalculate mean and length of each sample
+    lengths = np.asarray([len(arg) for arg in args])
+    means = np.asarray([np.mean(arg) for arg in args])
+
+    # (1) determine standard error of the mean for each sample
+    standard_errors = [np.std(arg, ddof=1) / np.sqrt(length)
+                       for arg, length in zip(args, lengths)]
+
+    # (2) define a weight for each sample
+    weights = ((1 / np.square(standard_errors)) /
+               np.sum(1 / np.square(standard_errors)))
+
+    # (3) determine variance-weighted estimate of the common mean
+    var_w = np.sum(weights * means)
+
+    # (4) determine one-sample t statistic for each group
+    t_stats = (means - var_w)/standard_errors
+
+    # calculate parameters to be used in transformation
+    v = lengths - 1
+    a = v - .5
+    b = 48 * a**2
+    c = (a * np.log(1 + (t_stats ** 2)/v))**.5
+
+    # (8) perform a normalizing transformation on t statistic
+    z = (c + ((c**3 + 3*c)/b) -
+         ((4*c**7 + 33*c**5 + 240*c**3 + 855*c) /
+          (b**2*10 + 8*b*c**4 + 1000*b)))
+
+    # (9) calculate statistic
+    A = np.sum(np.square(z))
+
+    # "[the p value is determined from] central chi-square random deviates
+    # with n_i - 1 degrees of freedom". Alexander, Govern (94)
+    p = distributions.chi2.sf(A, len(args) - 1)
+    return AlexanderGovernResult(A, p)
+
+
+def _alexandergovern_input_validation(args):
+    if len(args) < 2:
+        raise TypeError(f"2 or more inputs required, got {len(args)}")
+
+    # the current intended behavior is that the samples are flattened into 1d
+    args = [np.ravel(np.asarray(arg, dtype=float)) for arg in args]
```

Comment:
Mentiont that this is what happens for multidimensional arrays, and test the behavior (against raveling before calling the function).

Reply:
Added this to the test's notes:

> Inputs that are not one dimensional will be flattened.